### DISCUSSION TITLE
Dave did you manage to get your Pro working?

### DISCUSSION THREAD

**Discussion by anothermort** [BottlecapDave/HomeAssistant-OctopusEnergy#852] [General]
It does look like the integration will be awesome with the Pro added, now you can do local access to the meter's.

https://forum.octopus.energy/t/for-the-pro-user/8453/2026

Calling API: get_meter_status
Ambient energy consumption is: Low

Calling API: get_meter_consumption
Current total consumption is 4744.911kWh and the instant demand is 0.0kW

LED blinked!

Calling API: get_meter_status
Ambient energy consumption is: Low

Calling API: get_meter_consumption
Current total consumption is 4744.911kWh and the instant demand is 0.0kW

Calling API: get_meter_status
Ambient energy consumption is: Low

Calling API: get_meter_consumption
Current total consumption is 4744.911kWh and the instant demand is -0.008kW

Hopefully your Pro is working?

Thanks for all your hard work.

Cheers

Adam

**Comment by BottlecapDave** [comment]
Unfortunately I'm still struggling - not sure if it's my meter or the device itself :(

The initial APIs look like a good start. Hopefully soon they'll make them accessible outside of the device (or release details on how to call them from another device). I've seen people setup web servers etc to get access, but you can't expect people to go through those hoops with configuring with HA (or something similar)

**Reply by anothermort** [reply]
That's a pity, it seems to be a lottery if he Pro works with the meter or not.

The API is nice being local. I have set up a webserver which seems to work, but you have a very good point, its very fiddly at the least. Using someone else's code I could never do that on my own.

Maybe this requires some one clever than me to create a script etc that can self instal on the Pro.  Making all the endpoints required available.  Hopefully this will come with time from the community or from Octopus updates on the device.

Fingers crossed your Pro starts doing what it should.
